talos.nf

```diff
@@ -0,0 +1,235 @@
+    // runs the hail small-variant filtering
+    publishDir 'results', mode: 'copy'
+
+    input:
+        path matrix_table
+        path panelapp_data
+        path pedigree
+        path clinvar
+        path checkpoint
+        env TALOS_CONFIG
+
+//     output:
+//         path '${params.cohort}_small_variants_labelled.vcf.bgz', emit: 'vcf'
+//         path '${params.cohort}_small_variants_labelled.vcf.bgz.tbi', emit: 'index'
+
+    output:
+        tuple \
+            path("${params.cohort}_small_variants_labelled.vcf.bgz"), \
+            path("${params.cohort}_small_variants_labelled.vcf.bgz.tbi")
+
+    // only write a checkpoint if we were given a path
+    script:
+        def checkpoint = checkpoint.name != 'NO_FILE' ? "--checkpoint ${checkpoint}" : ''
+
+    // unzip the ClinvArbitration data directory and MatrixTable
+    """
+    tar -zxf ${clinvar}
+    tar -zxf ${matrix_table}
+
+    RunHailFiltering \
+        --input ${params.cohort}_small_variants.mt \
+        --panelapp ${panelapp_data} \
+        --pedigree ${pedigree} \
+        --output ${params.cohort}_small_variants_labelled.vcf.bgz \
+        --clinvar clinvarbitration_data/clinvar_decisions.ht \
+        --pm5 clinvarbitration_data/clinvar_pm5.ht \
+        ${checkpoint}
+    """
+}
+
+process RunHailFilteringSV {
+    // runs the hail structural-variant filtering
+    // here we can have multiple files - should we derive output name from input?
+    publishDir 'results', mode: 'copy'
+
+    input:
+        path matrix_table
+        path panelapp_data
+        path pedigree
+
+    output:
+        path '${params.cohort}_small_variants.vcf.bgz', emit: 'vcf'
+        path '${params.cohort}_small_variants.vcf.bgz.tbi', emit: 'index'
+
+    """
+    RunHailFilteringSV \
+        --mt ${matrix_table} \
+        --panelapp ${panelapp_data} \
+        --pedigree ${pedigree} \
+        --vcf_out ${params.cohort}_small_variants.vcf.bgz
+    """
+}
+
+process ValidateMOI {
+    // process the labelled variants
+    publishDir params.output_dir, mode: 'copy'
+
+    input:
+        tuple path(labelled_vcf), path(labelled_vcf_index)
+        path panelapp
+        path pedigree
+        path hpo_panel_matches
+        env TALOS_CONFIG
+
+    output:
+        path"${params.cohort}_results.json"
+
+    """
+    ValidateMOI \
+        --labelled_vcf ${labelled_vcf} \
+        --panelapp ${panelapp} \
+        --pedigree ${pedigree} \
+        --participant_panels ${hpo_panel_matches} \
+        --output ${params.cohort}_results.json
+    """
+}
+
+workflow {
+    // existence of these files is necessary for starting the workflow
+    // we open them as a channel, and pass the channel through to the method
+    pedigree_channel = Channel.fromPath(params.pedigree)
+    hpo_file_channel = Channel.fromPath(params.hpo)
+
+    // turn the VCF into a MatrixTable
+    input_vcf = Channel.fromPath(params.annotated_vcf)
+    input_vcf_idx = Channel.fromPath("${params.annotated_vcf}.tbi")
```

Comment:
You can do a map to make a vcf and index in the same variable (I don't remember the exact syntax).

```
input_vcf_and_index = Channel.fromPath(params.annotated_vcf)
map{ it -> [file(it), file("${it}.tbi")]}
```

Reply:
I knew there must be something for this exact situation... I found FromFilePairs which creates a list of pairs from a glob... not really what I want